collections-csharp/HashtableInCSharp/HashtableInCSharp/HashtableDemo.cs

```diff
@@ -4,85 +4,99 @@ namespace HashtableInCSharp
 {
     public class HashtableDemo
     {
-        public Hashtable CreateHashTable()
+        public static Hashtable CreateEmptyHashTable()
         {
             var firstHashTable = new Hashtable();
 
+            return firstHashTable;
+        }
+
+        public static Hashtable CreateHashTableWithInitialCapacity()
+        {
             var secondHashTable = new Hashtable(100);
 
+            return secondHashTable;
+        }
+
+        public static Hashtable CreateHashTableFromDictionary()
+        {
             var dictionary = new Dictionary<int, User>()
             {
-                { 1, new User() { FirstName = "Sam", LastName = "Manalt"} },
-                { 2, new User() { FirstName = "Lone", LastName = "Costa"}},
-                { 3, new User() { FirstName = "Alberto", LastName = "Daci"} },
+                { 1, new User() { Id = 1, FirstName = "Rafa", LastName = "Lopez" } },
+                { 2, new User() { Id = 2, FirstName = "Michael", LastName = "Sam" } },
+                { 3, new User() { Id = 3, FirstName = "Sam", LastName = "Manalt"} },
+                { 4, new User() { Id = 4, FirstName = "Lone", LastName = "Costa"} },
+                { 5, new User() { Id = 5, FirstName = "Alberto", LastName = "Daci"} },
             };
             
-            var thirdHashTable = new Hashtable(dictionary);
+            var userHashTable = new Hashtable(dictionary);
 
-            return secondHashTable;
+            return userHashTable;
         }
 
-        public Hashtable AddToHashTable(Hashtable hashTable)
+        public static Hashtable AddSampleDataToHashTable(Hashtable userHashTable, List<User> userList)
```

Comment:
This comment applies to all of the methods that take a Hashtable as an argument. Since each of the methods is actually modifying the passed in Hashtable, it is confusing to also have it return that same Hashtable. It would make sense in a Fluent programming style, Builder type of pattern, but since the main idea of this class is methods showing how to perform specific operations on a Hashtable, I think making these methods void would be make the intent clearer. It would also help to clarify the unit tests where the operation is being called and the return value is being captured and then the assertions are being made on the return value. It could lead to confusion as to thinking the operations are not modifying the original, but rather returning a copy that has had the operations applied to it (like you have with the elements of the Immutable collections library).